Client/mods/deathmatch/logic/luadefs/CLuaWorldDefs.cpp

```diff
@@ -242,18 +252,38 @@ int CLuaWorldDefs::ProcessLineOfSight(lua_State* luaVM)
     argStream.ReadBool(flags.bSeeThroughStuff, false);
     argStream.ReadBool(flags.bIgnoreSomeObjectsForCamera, false);
     argStream.ReadBool(flags.bShootThroughStuff, false);
-    argStream.ReadUserData(pIgnoredElement, NULL);
+
+    if (argStream.NextIsTable()) // Is the next value a table? Read it as a user data table (will error if table contains invalid type)
+    {
+        argStream.ReadUserDataTable(vecIgnoredElements);
+    }
+    else if (argStream.NextIsNil()) // Is the next value nil? We can just skip to reading bIncludeBuildingInfo
```

Comment:
`NextIsNone()` I'd instead I'd say. Nil is error prone (user thinks he's passing in an element, and is wondering why it's not working)

Reply:
I need to check for `NextIsNil` to keep the default behaviour. 

If passing nil, it would would skip to the next arg.

If passing none, then it doesn't need to skip because nothing has been passed, so there also can't be another argument after that.

Honestly, I did like an hour or two of testing conditions on this and I'm certain my version is the most consistent and keeps the previous behaviour. If you can provide me with the exact logic you think would work better, I would be happy to change it.

Reply:
There can be an argument after a none. If you pass `_` it act's as none, not nil.

Reply:
I'm sure I tested that and it was nil, I'll try it again later. If it does work how you suggest then I'll just change the `else if (argStream.NextIsNil())` to `else if (argStream.NextIsNone())` - that should be fine.

Reply:
I have removed vararg for this, it's kind of unnecessary. It'll accept either single element (current behaviour) or table of elements.